src/components/SupportRequestModal.js

```diff
@@ -14,91 +13,96 @@ import validate from 'validate.js'
 
 
 const constraints = {
-  email: { email: true },
+  name: { presence: { allowEmpty: false } },
   subject: { presence: { allowEmpty: false } },
-  description: { presence: { allowEmpty: false } }
+  description: { presence: { allowEmpty: false } },
+  email: { email: true, presence: { allowEmpty: false } }
 }
 
-const SupportRequestModal = _.flow(
-  ajaxCaller,
-  Utils.connectAtom(authStore, 'authState')
-)(class SupportRequestModal extends Component {
-  constructor(props) {
-    super(props)
-    const { contactEmail, email } = props.authState.profile
+const SupportRequestModal =
+  Utils.connectAtom(authStore, 'authState')(class SupportRequestModal extends Component {
+    constructor(props) {
+      super(props)
+      const { contactEmail, email } = props.authState.profile
 
-    this.state = {
-      subject: '',
-      description: '',
-      type: 'question',
-      email: contactEmail || email
+      this.state = {
+        subject: '',
+        description: '',
+        type: 'question',
+        email: contactEmail || email || '',
+        nameEntered: ''
+      }
     }
-  }
 
-  render() {
-    const { onDismiss, authState: { profile: { firstName } } } = this.props
-    const { submitting, submitError, subject, description, type, email } = this.state
-    const greetUser = firstName === 'N/A' ? `?` : `, ${firstName}?`
-    const errors = validate({ email, description, subject }, constraints)
+    render() {
+      const { onDismiss, authState: { isSignedIn, profile: { firstName } } } = this.props
+      const { submitting, submitError, subject, description, type, email, nameEntered } = this.state
+      const greetUser = isSignedIn ? `, ${firstName}` : ``
+      const errors = validate({ email, description, subject, name: isSignedIn ? firstName : nameEntered }, constraints)
```

Comment:
Is it possible for someone's profile name to be blank? If so, they could end up being unable to submit.

Reply:
That's a good point that if we can't load the user profile info that we should give them the option of filling in their name manually. I might change the Name field to display if a user is not signed in, or if they don't have a name. I think that would allow them to still submit feedback even if our dev service is down.